runtime/litmus/src/tests/xcm_parachain/mod.rs

```diff
@@ -14,1148 +14,96 @@
 // You should have received a copy of the GNU General Public License
 // along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
 
-use super::setup::*;
+use runtime_common::tests::{xcm_parachain, xcm_parachain::TestXCMRequirements};
+
 use crate::{
-	xcm_config::{
-		CurrencyId, CurrencyIdMultiLocationConvert, LocationToAccountId, UnitWeightCost,
-		XcmFeesAccount,
+	tests::{
+		Call as RelayCall, Origin as RelayOrigin, ParaA, ParaB, Relay, RelayChainRuntime, TestNet,
 	},
-	Origin,
-};
-use cumulus_primitives_core::{ParaId, PersistedValidationData};
-use cumulus_primitives_parachain_inherent::ParachainInherentData;
-
-use codec::Encode;
-use frame_support::{
-	assert_noop, assert_ok,
-	traits::{Currency, PalletInfoAccess},
+	xcm_config::{LocationToAccountId, UnitWeightCost},
+	Call, Origin, Runtime,
 };
-use frame_system::RawOrigin;
-use orml_traits::MultiCurrency;
-use polkadot_parachain::primitives::RelayChainBlockNumber;
-use sp_runtime::{traits::Convert, AccountId32};
-use xcm::prelude::*;
-use xcm_executor::traits::Convert as xcmConvert;
-use xcm_simulator::TestExt;
 
-pub mod relay_sproof_builder;
+// pub mod relay_sproof_builder;
+// pub const RELAY_UNIT: u128 = 1;
 
-pub const RELAY_UNIT: u128 = 1;
+struct LitmusTestXCMRequirements;
 
-fn para_account(x: u32) -> AccountId32 {
-	<relay::SovereignAccountOf as xcmConvert<MultiLocation, AccountId32>>::convert(
-		Parachain(x).into(),
-	)
-	.unwrap()
-}
-
-fn sibling_account(x: u32) -> AccountId32 {
-	<LocationToAccountId as xcmConvert<MultiLocation, AccountId32>>::convert(
-		(Parent, Parachain(x)).into(),
-	)
-	.unwrap()
-}
+impl TestXCMRequirements for LitmusTestXCMRequirements {
```

Comment:
If I try to be picky here:
we can even put these trait binding into macro, as well as the following common test cases, so eventually we'll only have something like in this file
```
run_xcm_tests!(LitmusTestXCMRequirements);
```
or maybe we don't even need `LitmusTestXCMRequirements` if the runtime binding is the same for all chains

Reply:
Extract common tests into macros, inclue: `base_call_filter` `transaction_payment` `xcm_parachain`